src/cli.rs

```diff
@@ -508,6 +514,9 @@ fn get_actual_app_id(
     path: &Option<String>,
 ) -> Result<String, CommandError> {
     match app_id {
+        Some(id) if id.is_empty() => {
+            Err(CommandError::EmptyAppId("this".to_string()))
```

Comment:
what's "this?"?

Reply:
Hmm. Seems to be not very obviuous. So the only command that actually needs it is upload.

Because we should not pass empty app id to function. Rather when handling CLI commands we should show an early error.

Like I mentioned to you before you need to modify get_actual_app_id to return Error. It should return an error when app_id ends up being empty.

The only one that has to be outside of this is upload because it reads the whole manifest (we could move reading manifest to CLI handling but I am ok with it staying as it is for now)

Reply:
> Hmm. Seems to be not very obviuous. So the only command that actually needs it is upload.
> 
> Because we should not pass empty app id to function. Rather when handling CLI commands we should show an early error.
> 
> Like I mentioned to you before you need to modify get_actual_app_id to return Error. It should return an error when app_id ends up being empty.

Well, that's exactly what I did - first case in match throws error if the `app_id` provided by the user is empty, then if `app_id` is empty in the manifest file, a separate error is being thrown. 

> 
> The only one that has to be outside of this is upload because it reads the whole manifest (we could move reading manifest to CLI handling but I am ok with it staying as it is for now)

Reply:
@StanIwan sorry I can't find it for other commands like promote - can you show me where they handle that?

Reply:
Ok I see. Does it work properly? Did you test it with all commands?

Reply:
Also you can remove "this" string. I don't see the point - we always know which command we call. You can always match if it's a MissingAppId error and show that.

Reply:
I tested it with some, can test it with all

Reply:
Now, I tested it with most commands and works the same every time. Also, updated error string.